### ISSUE TITLE
Failing to connect to server

### ISSUE THREAD

**Issue by Kairus101** [ammaraskar/pyCraft#65]
First time attempting to run this, the latest src gives

```
Traceback (most recent call last):
  File "D:\Python\Python36\lib\threading.py", line 916, in _bootstrap_inner
    self.run()
  File "D:\Users\Kairus\Downloads\pyCraft-master\minecraft\networking\connection.py", line 341, in run
    self.connection._handle_exception(e, sys.exc_info())
  File "D:\Users\Kairus\Downloads\pyCraft-master\minecraft\networking\connection.py", line 319, in _handle_exception
    raise_(*exc_info)
  File "D:\Python\Python36\lib\site-packages\future\utils\__init__.py", line 413, in raise_
    raise exc.with_traceback(tb)
  File "D:\Users\Kairus\Downloads\pyCraft-master\minecraft\networking\connection.py", line 339, in run
    self._run()
  File "D:\Users\Kairus\Downloads\pyCraft-master\minecraft\networking\connection.py", line 371, in _run
    self.connection.file_object)
  File "D:\Users\Kairus\Downloads\pyCraft-master\minecraft\networking\connection.py", line 424, in read_packet
    ready_to_read = select.select([stream], [], [], self.TIME_OUT)[0]
TypeError: argument must be an int, or have a fileno() method.

Bye!
```

I also tried commit 8d1dcec - the 1.12 release
```
Traceback (most recent call last):
  File "D:\Python\Python36\lib\threading.py", line 916, in _bootstrap_inner
    self.run()
  File "D:\Users\Kairus\Downloads\pyCraft-master\minecraft\networking\connection.py", line 341, in run
    self.connection._handle_exception(e, sys.exc_info())
  File "D:\Users\Kairus\Downloads\pyCraft-master\minecraft\networking\connection.py", line 319, in _handle_exception
    raise_(*exc_info)
  File "D:\Python\Python36\lib\site-packages\future\utils\__init__.py", line 413, in raise_
    raise exc.with_traceback(tb)
  File "D:\Users\Kairus\Downloads\pyCraft-master\minecraft\networking\connection.py", line 339, in run
    self._run()
  File "D:\Users\Kairus\Downloads\pyCraft-master\minecraft\networking\connection.py", line 371, in _run
    self.connection.file_object)
  File "D:\Users\Kairus\Downloads\pyCraft-master\minecraft\networking\connection.py", line 439, in read_packet
    decompressed_packet = decompress(packet_data.read())
zlib.error: Error -5 while decompressing data: incomplete or truncated stream
```

Python -- version
Python 3.6.2
pip install -r requirements.txt
```
Requirement already satisfied: cryptography>=1.5 in d:\python\python36\lib\site-packages (from -r requirements.txt (line 1))
Requirement already satisfied: requests in d:\python\python36\lib\site-packages(from -r requirements.txt (line 2))
Requirement already satisfied: future in d:\python\python36\lib\site-packages (from -r requirements.txt (line 3))
Requirement already satisfied: idna>=2.1 in d:\python\python36\lib\site-packages (from cryptography>=1.5->-r requirements.txt (line 1))
Requirement already satisfied: cffi>=1.7 in d:\python\python36\lib\site-packages (from cryptography>=1.5->-r requirements.txt (line 1))
Requirement already satisfied: six>=1.4.1 in d:\python\python36\lib\site-packages (from cryptography>=1.5->-r requirements.txt (line 1))
Requirement already satisfied: asn1crypto>=0.21.0 in d:\python\python36\lib\site-packages (from cryptography>=1.5->-r requirements.txt (line 1))
Requirement already satisfied: urllib3<1.22,>=1.21.1 in d:\python\python36\lib\site-packages (from requests->-r requirements.txt (line 2))
Requirement already satisfied: certifi>=2017.4.17 in d:\python\python36\lib\site-packages (from requests->-r requirements.txt (line 2))
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in d:\python\python36\lib\site-packages (from requests->-r requirements.txt (line 2))
Requirement already satisfied: pycparser in d:\python\python36\lib\site-packages (from cffi>=1.7->cryptography>=1.5->-r requirements.txt (line 1))
```

Any insight? I'm happy to help in working this out.

**Comment by joodicator** [comment]
The first error is due to a regression in a recent commit (c87d7bc6f3a06cdb27134fc713e8163693068e81). I have reverted this in cab8d56746355b158087ed64ad067d66e07f8f31.

The regression wasn't noticed because our test suite currently only does a full connection test without encryption (and without compression, in offline mode). I think we will have to add encryption and compression tests, to address this.

I do not know the cause of the second error, but I suspect that if it happens with the 1.12 commit, it will also happen even after the latest fix. @Kairus101 Can I ask what kind of server you're connecting to, and if it has any unusual configuration?

**Comment by Kairus101** [comment]
Nope, that fixed it - not entirely sure what was wrong with the initial 1.12 commit, though this particular bug was fixed with [cab8d56](https://github.com/ammaraskar/pyCraft/commit/cab8d56746355b158087ed64ad067d66e07f8f31). Thanks!
